### ISSUE TITLE
[Accessibility][Sourcing]: fdp-table accessibility issues

### ISSUE THREAD

**Issue by I543348** [SAP/fundamental-ngx#9135]
#### Is this a bug, enhancement, or feature request?
Bug

#### Briefly describe your proposal.
1) While announcing table cell value - Its corresponding column header is not output by screenreader
2) All tables have role=table, this should be changed as grid and treegrid accordingly: https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/
3) Enter and exit edit mode not available when there are more than one interactive elements inside a table cell

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
Angular 13.x.x
   "@fundamental-ngx/core": "0.35.4",
    "@fundamental-ngx/platform": "0.35.4",

#### If this is a bug, please provide steps for reproducing it.
Can be reproducible on https://sap.github.io/fundamental-ngx/#/platform/table 

#### Is there anything else we should know?
This is a blocker for guided sourcing customers

**Comment by I543348** [comment]
This works fine , when there are more than 1 interactive elements in a table call , ESC key is used to come out of interactive elements.

**Comment by mikerodonnell89** [comment]
Reopening for "description" being read twice

**Comment by fkolar** [comment]
I think this is more a duplicate of #9527. I think this is the same problem. we should keep only one issue. 
@I543348

**Comment by mikerodonnell89** [comment]
To elaborate - column header is read twice for jaws only when first accessing a cell in the grid for the first time ("Name" seen here twice after clicking "Astro Phone 6". The column heading is only read once after clicking "Beam Breaker B-1"
![Screenshot 2023-05-10 at 4 30 57 PM](https://github.com/SAP/fundamental-ngx/assets/2471874/79b6c1c3-6981-424e-87c0-53d545f5c115)

**Comment by I543348** [comment]
@mikerodonnell89 , We checked with Accessibility Test Lab (ATL) team as eve for first time , the column header should not be read twice.
This has to be reopened

**Comment by I543348** [comment]
Hi Team, i just checked with Central accessibility team and they are gave below inputs.
Please check and help.

<img width="846" alt="Screenshot 2023-06-01 at 10 45 51 PM" src="https://github.com/SAP/fundamental-ngx/assets/88182358/bd1fe4c4-8262-432e-92b2-e1eae6303e5e">

cc: @mikerodonnell89 , @fkolar

**Comment by I543348** [comment]
This is working fine.
Thanks.
